api/src/services/survey-service.ts

```diff
@@ -142,7 +142,14 @@ export class SurveyService extends DBService {
    * @memberof SurveyService
    */
   async getSurveyData(surveyId: number): Promise<GetSurveyData> {
-    return this.surveyRepository.getSurveyData(surveyId);
+    const [surveyData, surveyTypesData] = await Promise.all([
```

Comment:
Is there any reason we prefer `Promise.all` over something like:
```typescript
const surveyData = await this.surveyRepository.getSurveyData(surveyId);
const surveyTypesData = await this.surveyRepository.getSurveyTypesData(surveyId);
```
I'm guessing it's because using `Promise.all` allows both these promises to be fullfilled concurrently?

Reply:
Yeah just being able to run both at the same time, thats all.